### PR TITLE
Bug 1378470 - Run XCUITests on iPad as well

### PR DESCRIPTION
**Pull request by ugarlapati**
…ed the code to support both.

### CODE REVIEW

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
BaseTestCase.swift | 75 | if,for,while,do statements shouldn't wrap their conditionals in parentheses. 
#### Errors

File | Line | Reason |
| --- | ----- | ----- |
BaseTestCase.swift | 75 | Opening braces should be preceded by a single space and on the same line as the declaration. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: XCUITests/PrivateBrowsingTest.swift [hunk 1]
```diff
@@ -99,28 +129,59 @@ class PrivateBrowsingTest: BaseTestCase {
...
-        navigator.openURL(urlString: url1)
-        navigator.nowAt(PrivateBrowserTab)
-        navigator.goto(PrivateTabTray)
-
-        // Go back to regular browser
-        navigator.goto(TabTray)
-
-        // Go back to private browsing and check that the tab has not been closed
-        navigator.goto(PrivateTabTray)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/PrivateBrowsingTest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/PrivateBrowsingTest.swift [hunk 2]
```diff
@@ -99,28 +129,59 @@ class PrivateBrowsingTest: BaseTestCase {
...
+            navigator.goto(PrivateTabTray)
+        }
         waitforExistence(app.collectionViews.cells[url1Label])
         let numPrivTabs = app.collectionViews.cells.count
         XCTAssertEqual(numPrivTabs, 1, "The number of tabs is not correct, the private tab should not have been closed")
 
         // Now the enable the Close Private Tabs when closing the Private Browsing Button
-        navigator.goto(SettingsScreen)
+       
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/PrivateBrowsingTest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/PrivateBrowsingTest.swift [hunk 3]
```diff
@@ -99,28 +129,59 @@ class PrivateBrowsingTest: BaseTestCase {
...
+        } else {
+            navigator.goto(SettingsScreen)
+        }
         closePrivateTabsSwitch.tap()
 
         // Go back to regular browsing and check that the private tab has been closed and that the initial Private Browsing message appears when going back to Private Browsing
-        navigator.goto(PrivateTabTray)
-        navigator.goto(TabTray)
-        navigator.goto(PrivateTabTray)
+        if isiPad() {
```
Comment (npark-mozilla) [inline XCUITests/PrivateBrowsingTest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/PrivateBrowsingTest.swift [hunk 4]
```diff
@@ -130,20 +191,39 @@ class PrivateBrowsingTest: BaseTestCase {
 
     func testPrivateBrowserPanelView() {
         // If no private tabs are open, there should be a initial screen with label Private Browsing
-        navigator.goto(PrivateTabTray)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/PrivateBrowsingTest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/PrivateBrowsingTest.swift [hunk 5]
```diff
@@ -130,20 +191,39 @@ class PrivateBrowsingTest: BaseTestCase {
...
+            navigator.goto(PrivateTabTray)
+        }
 
         XCTAssertTrue(app.staticTexts["Private Browsing"].exists, "Private Browsing screen is not shown")
         let numPrivTabsFirstTime = app.collectionViews.cells.count
         XCTAssertEqual(numPrivTabsFirstTime, 0, "The number of tabs is not correct, there should not be any private tab yet")
 
         // If a private tab is open Private Browsing screen is not shown anymore
-        navigator.goto(NewPrivateTabScreen)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/PrivateBrowsingTest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/PrivateBrowsingTest.swift [hunk 6]
```diff
@@ -130,20 +191,39 @@ class PrivateBrowsingTest: BaseTestCase {
...
+            app.buttons["TabTrayController.addTabButton"].tap()
+        } else {
+            navigator.goto(NewPrivateTabScreen)
+        }
 
         // Go to regular browsing
-        navigator.goto(TabTray)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/PrivateBrowsingTest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/PrivateBrowsingTest.swift [hunk 7]
```diff
@@ -130,20 +191,39 @@ class PrivateBrowsingTest: BaseTestCase {
...
+            app.buttons["TopTabsViewController.tabsButton"].tap()
+            app.buttons["TabTrayController.maskButton"].tap()
+        } else {
+            navigator.goto(TabTray)
+        }
 
         // Go back to private brosing
-        navigator.goto(PrivateTabTray)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/PrivateBrowsingTest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/ReaderViewUITest.swift [hunk 8]
```diff
@@ -51,9 +51,18 @@ class ReaderViewTest: BaseTestCase {
         navigator.nowAt(NewTabScreen)
         // Add item to reading list and check that it appears
         addContentToReaderView()
-        navigator.goto(NewTabScreen)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/ReaderViewUITest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/ReaderViewUITest.swift [hunk 9]
```diff
@@ -51,9 +51,18 @@ class ReaderViewTest: BaseTestCase {
...
+            app.buttons["TopTabsViewController.tabsButton"].tap()
+            app.buttons["TabTrayController.addTabButton"].tap()
+        } else {
+            navigator.goto(NewTabScreen)
+        }
         waitforExistence(app.buttons["HomePanels.ReadingList"])
-        navigator.goto(HomePanel_ReadingList)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/ReaderViewUITest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/ReaderViewUITest.swift [hunk 10]
```diff
@@ -85,9 +94,19 @@ class ReaderViewTest: BaseTestCase {
         waitforExistence(app.buttons["Add to Reading List"])
 
         // Go to reader list view to check that there is not any item there
-        navigator.goto(NewTabScreen)
+        
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/ReaderViewUITest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/ReaderViewUITest.swift [hunk 11]
```diff
@@ -97,9 +116,18 @@ class ReaderViewTest: BaseTestCase {
 
     func testMarkAsReadAndUnreadFromReadingList() {
         addContentToReaderView()
-        navigator.goto(NewTabScreen)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/ReaderViewUITest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/ReaderViewUITest.swift [hunk 12]
```diff
@@ -97,9 +116,18 @@ class ReaderViewTest: BaseTestCase {
...
+            app.buttons["TopTabsViewController.tabsButton"].tap()
+            app.buttons["TabTrayController.addTabButton"].tap()
+        } else {
+            navigator.goto(NewTabScreen)
+        }
         waitforExistence(app.buttons["HomePanels.ReadingList"])
-        navigator.goto(HomePanel_ReadingList)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/ReaderViewUITest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/ReaderViewUITest.swift [hunk 13]
```diff
@@ -115,9 +143,18 @@ class ReaderViewTest: BaseTestCase {
 
     func testRemoveFromReadingList() {
         addContentToReaderView()
-        navigator.goto(NewTabScreen)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/ReaderViewUITest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/ReaderViewUITest.swift [hunk 14]
```diff
@@ -115,9 +143,18 @@ class ReaderViewTest: BaseTestCase {
...
+            app.buttons["TopTabsViewController.tabsButton"].tap()
+            app.buttons["TabTrayController.addTabButton"].tap()
+        } else {
+            navigator.goto(NewTabScreen)
+        }
         waitforExistence(app.buttons["HomePanels.ReadingList"])
-        navigator.goto(HomePanel_ReadingList)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/ReaderViewUITest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/ThirdPartySearchTest.swift [hunk 15]
```diff
@@ -26,108 +26,115 @@ class ThirdPartySearchTest: BaseTestCase {
...
-        tabTrayButton(forApp: app).tap()
-        app.buttons["TabTrayController.addTabButton"].tap()
-        app.textFields["url"].tap()
-        app.typeText("window")
-        app.scrollViews.otherElements.buttons["developer.mozilla.org search"].tap()
-
-        // Ensure that the search is done on MDN
-        let url = app.textFields["url"].value as! String
-        XCTAssert(url.hasPrefix("https://developer.mozilla.org/en-US/search"), "The URL should indicate that the search was performed on MDN and not the default")
+        if isiPad() == false {
```
Comment (npark-mozilla) [inline XCUITests/ThirdPartySearchTest.swift]: !isiPad()

File: XCUITests/TopTabsTest.swift [hunk 16]
```diff
@@ -39,15 +39,25 @@ class TopTabsTest: BaseTestCase {
     }
 
     func testAddTabFromTabTray() {
-        navigator.goto(TabTray)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/TopTabsTest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/TopTabsTest.swift [hunk 17]
```diff
@@ -39,15 +39,25 @@ class TopTabsTest: BaseTestCase {
...
+        }
         navigator.openURL(urlString: url)
         waitForValueContains(app.textFields["url"], value: urlValue)
         // The tabs counter shows the correct number
         let tabsOpen = app.buttons["Show Tabs"].value
         XCTAssertEqual("2", tabsOpen as? String)
 
         // The tab tray shows the correct tabs
-        navigator.goto(TabTray)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/TopTabsTest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/TopTabsTest.swift [hunk 18]
```diff
@@ -65,7 +75,11 @@ class TopTabsTest: BaseTestCase {
         //app.buttons["Switch"].tap(force: true)
 
         // Open tab tray to check that both tabs are there
-        navigator.goto(TabTray)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/TopTabsTest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/TopTabsTest.swift [hunk 19]
```diff
@@ -75,24 +89,43 @@ class TopTabsTest: BaseTestCase {
     func testSwitchBetweenTabs() {
         // Open two urls from tab tray and switch between them
         navigator.openURL(urlString: url)
-        navigator.goto(TabTray)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/TopTabsTest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/TopTabsTest.swift [hunk 20]
```diff
@@ -75,24 +89,43 @@ class TopTabsTest: BaseTestCase {
...
+            app.buttons["TopTabsViewController.tabsButton"].tap()
+            app.buttons["TabTrayController.addTabButton"].tap()
+        } else {
+            navigator.goto(TabTray)
+        }
         navigator.openURL(urlString: urlYah)
-        navigator.goto(TabTray)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/TopTabsTest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/TopTabsTest.swift [hunk 21]
```diff
@@ -75,24 +89,43 @@ class TopTabsTest: BaseTestCase {
...
+        } else {
+            navigator.goto(TabTray)
+        }
         waitforExistence(app.collectionViews.cells[urlLabel])
         app.collectionViews.cells[urlLabel].tap()
         waitForValueContains(app.textFields["url"], value: urlValue)
 
         navigator.nowAt(BrowserTab)
-        navigator.goto(TabTray)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/TopTabsTest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/TopTabsTest.swift [hunk 22]
```diff
@@ -75,24 +89,43 @@ class TopTabsTest: BaseTestCase {
...
         waitforExistence(app.collectionViews.cells[urlLabelYah])
         app.collectionViews.cells[urlLabelYah].tap()
         waitForValueContains(app.textFields["url"], value: urlValueYah)
     }
 
     func testCloseOneTab() {
         navigator.openURL(urlString: url)
-        navigator.goto(NewTabScreen)
-        navigator.goto(TabTray)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/TopTabsTest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/TopTabsTest.swift [hunk 23]
```diff
@@ -109,17 +142,35 @@ class TopTabsTest: BaseTestCase {
         // A different tab than home is open to do the proper checks
         navigator.openURL(urlString: url)
         // Add several tabs and check that the number is correct
-        navigator.createSeveralTabsFromTabTray (numberTabs: 3)
-        navigator.goto(TabTray)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/TopTabsTest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/TopTabsTest.swift [hunk 24]
```diff
@@ -109,17 +142,35 @@ class TopTabsTest: BaseTestCase {
...
+            navigator.goto(TabTray)
+        }
 
         waitforExistence(app.collectionViews.cells[urlLabel])
         let numTabsOpen = app.collectionViews.cells.count
         XCTAssertEqual(numTabsOpen, 4, "The number of regular tabs is not correct")
 
         // Close all tabs, undo it and check that the number of tabs is correct
-        navigator.closeAllTabs()
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/TopTabsTest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/TopTabsTest.swift [hunk 25]
```diff
@@ -130,16 +181,30 @@ class TopTabsTest: BaseTestCase {
         // A different tab than home is open to do the proper checks
         navigator.openURL(urlString: url)
         // Add several tabs from tab tray menu and check that the  number is correct before closing all
-        navigator.createSeveralTabsFromTabTray (numberTabs: 3)
-        navigator.goto(TabTray)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/TopTabsTest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

File: XCUITests/TopTabsTest.swift [hunk 26]
```diff
@@ -130,16 +181,30 @@ class TopTabsTest: BaseTestCase {
...
+        }
 
         waitforExistence(app.collectionViews.cells[urlLabel])
         let numTabsOpen = app.collectionViews.cells.count
         XCTAssertEqual(numTabsOpen, 4, "The number of tabs is not correct after opening 4")
 
         // Close all tabs and check that the number of tabs is correct
-        navigator.closeAllTabs()
-        navigator.goto(TabTray)
+        if isiPad() == true {
```
Comment (npark-mozilla) [inline XCUITests/TopTabsTest.swift]: this if statement can be removed as well. fxscreengraph can handle this.

**Comment by npark-mozilla** [comment]
Also, let's update the code according to the SwiftLint result above.

File: XCUITests/FindInPageTest.swift [hunk 27]
```diff
@@ -86,7 +86,11 @@ class FindInPageTests: BaseTestCase {
 
         // Going to tab tray and back to the website hides the search field.
         navigator.nowAt(BrowserTab)
-        navigator.goto(TabTray)
+        if isiPad() == true {
```
Comment (ugarlapati) [inline XCUITests/FindInPageTest.swift]: navigator.goto(TabTray) method is not working in the above test case for iPad.

File: XCUITests/HomePageSettingsUITest.swift [hunk 28]
```diff
@@ -104,7 +112,14 @@ class HomePageSettingsUITests: BaseTestCase {
 
         // Go to HomePage settings and check that it is not possible to copy it into the set webpage field
         navigator.nowAt(BrowserTab)
-        navigator.goto(HomePageSettings)
+        if isiPad() == true {
```
Comment (ugarlapati) [inline XCUITests/HomePageSettingsUITest.swift]: For iPad its failing if use the fxscreengraph methods.

File: XCUITests/HomePageSettingsUITest.swift [hunk 29]
```diff
@@ -82,9 +82,17 @@ class HomePageSettingsUITests: BaseTestCase {
         app.textFields["url"].press(forDuration: 5)
         app.buttons["Copy Address"].tap()
 
+        
         // Go to HomePage settings and paste it using the option Used Copied Link
-        navigator.goto(SettingsScreen)
-        navigator.goto(HomePageSettings)
+        if isiPad() == true {
```
Comment (ugarlapati) [inline XCUITests/HomePageSettingsUITest.swift]: For iPad its failing if use the fxscreengraph methods.

File: XCUITests/PrivateBrowsingTest.swift [hunk 30]
```diff
@@ -33,55 +33,85 @@ class PrivateBrowsingTest: BaseTestCase {
...
         // History without counting Recently Closed and Synced devices
         let history = app.tables["History List"].cells.count - 2
-
+        
         XCTAssertEqual(history, 1, "History entries in regular browsing do not match")
-
+        
         // Go to Private browsing to open a website and check if it appears on History
-        navigator.goto(PrivateTabTray)
+        if isiPad() == true {
```
Comment (ugarlapati) [inline XCUITests/PrivateBrowsingTest.swift]: For iPad its failing if use the fxscreengraph methods.

File: XCUITests/PrivateBrowsingTest.swift [hunk 31]
```diff
@@ -33,55 +33,85 @@ class PrivateBrowsingTest: BaseTestCase {
...
         XCTAssertEqual(privateHistory, 1, "History entries in private browsing do not match")
     }
 
     func testTabCountShowsOnlyNormalOrPrivateTabCount() {
         // Open two tabs in normal browsing and check the number of tabs open
         navigator.openNewURL(urlString: url1)
-        navigator.goto(TabTray)
-        navigator.goto(NewTabScreen)
-        navigator.goto(TabTray)
+        if isiPad() == true {
```
Comment (ugarlapati) [inline XCUITests/PrivateBrowsingTest.swift]: For iPad its failing if use the fxscreengraph methods.

File: XCUITests/PrivateBrowsingTest.swift [hunk 32]
```diff
@@ -33,55 +33,85 @@ class PrivateBrowsingTest: BaseTestCase {
...
+            navigator.goto(TabTray)
+        }
 
         waitforExistence(app.collectionViews.cells[url1Label])
         let numTabs = app.collectionViews.cells.count
         XCTAssertEqual(numTabs, 2, "The number of regular tabs is not correct")
 
         // Open one tab in private browsing and check the total number of tabs
-        navigator.goto(NewPrivateTabScreen)
+        if isiPad() == true {
```
Comment (ugarlapati) [inline XCUITests/PrivateBrowsingTest.swift]: For iPad its failing if use the fxscreengraph methods.

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
ClipBoardTests.swift | 61 | Limit vertical whitespace to a single empty line. Currently 2. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**